pkg/scheduler/framework/plugins/volumebinding/volume_binding.go

```diff
@@ -118,37 +126,79 @@ func (pl *VolumeBinding) EventsToRegister() []framework.ClusterEventWithHint {
 		// When CSIStorageCapacity is enabled, pods may become schedulable
 		// on CSI driver & storage capacity changes.
 		{Event: framework.ClusterEvent{Resource: framework.CSIDriver, ActionType: framework.Add | framework.Update}},
-		{Event: framework.ClusterEvent{Resource: framework.CSIStorageCapacity, ActionType: framework.Add | framework.Update}},
+		{Event: framework.ClusterEvent{Resource: framework.CSIStorageCapacity, ActionType: framework.Add | framework.Update}, QueueingHintFn: pl.isSchedulableAfterCSIStorageCapacityChange},
 	}
 	return events
 }
 
+func (pl *VolumeBinding) isSchedulableAfterCSIStorageCapacityChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
+	oldCap, newCap, err := util.As[*storagev1.CSIStorageCapacity](oldObj, newObj)
+	if err != nil {
+		return framework.Queue, err
+	}
+
+	logger = klog.LoggerWithValues(
+		logger,
+		"Pod", klog.KObj(pod),
+		"CSIStorageCapacity", klog.KObj(newCap),
+	)
+
+	pinfoList := getPVCInfoListFromPodVolumes(pod)
+	for _, pinfo := range pinfoList {
+		pvc, err := pl.PVCLister.PersistentVolumeClaims(pod.Namespace).Get(pinfo.pvcName)
+		if err != nil {
+			if pinfo.isEphemeral && apierrors.IsNotFound(err) {
+				err = fmt.Errorf("waiting for ephemeral volume controller to create the persistentvolumeclaim %q", pinfo.pvcName)
+				return framework.Queue, err
+			}
+			return framework.Queue, err
+		}
+
+		className := volume.GetPersistentVolumeClaimClass(pvc)
+		if className == "" {
```

Comment:
As described in [Reserving a PersistentVolume](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#reserving-a-persistentvolume), you can create a PVC with an empty StorageClass, for example, if you are binding an existing PV to the other PVC. However, since the PV is already provisioned and changes to the CSIStorageCapacity do not affect the scheduling of the Pod, this should not matter, right?

Reply:
[Note for self] If StorageClass is not specified, it seems to be automatically set.
[https://github.com/kubernetes/kubernetes/blob/c0f4879164f3d52dc19e49a7419bc076c66832d2/pkg/controller/volume/persistentvolume/pv_controller.go#L930-L958](https://github.com/kubernetes/kubernetes/blob/c0f4879164f3d52dc19e49a7419bc076c66832d2/pkg/controller/volume/persistentvolume/pv_controller.go#L930-L958)

Reply:
[Note for self] When binding an existing PV to another PVC, `"pv.kubernetes.io/bound-by-controller": "yes"` is set.
[https://github.com/kubernetes/kubernetes/blob/c0f4879164f3d52dc19e49a7419bc076c66832d2/pkg/controller/volume/persistentvolume/pv\_controller.go#L1009-L1034](https://github.com/kubernetes/kubernetes/blob/c0f4879164f3d52dc19e49a7419bc076c66832d2/pkg/controller/volume/persistentvolume/pv%5C_controller.go#L1009-L1034)

Reply:
[Note for self] When `"pv.kubernetes.io/bound-by-controller": "yes"` is set, `syncUnboundClaim`, which calls `assignDefaultStorageClass`, is not executed.

https://github.com/kubernetes/kubernetes/blob/c0f4879164f3d52dc19e49a7419bc076c66832d2/pkg/controller/volume/persistentvolume/pv_controller.go#L249-L253

Reply:
I've modified the code to handle the cases you pointed out.
Additionally, I realized that I had not considered the case of using the default StorageClass, so I have also addressed that.
https://github.com/kubernetes/kubernetes/pull/124961/commits/a76c39685884c96e0b861d27859ce0d4d1e75c43